.github/workflows/release_ci.yml

```diff
@@ -0,0 +1,320 @@
+          cp -R "./${TARGET_DIR}release/imageflow.pdb" ./artifacts/staging/imageflow.pdb
+          cp bindings/headers/*.h  ./artifacts/staging/headers/
+          cp bindings/headers/imageflow_default.h  ./artifacts/staging/imageflow.h
+          cp ./ci/packaging_extras/{install,uninstall}.sh ./artifacts/staging
+          rm ./artifacts/staging/*.{o,d,rlib} || true
+          rm ./artifacts/staging/*-* || true
+          (
+            cd ./artifacts/staging
+            7z a -r "./archive.zip" ./*
+          )
+          mkdir -p "./artifacts/upload/$(dirname "${ARTIFACT_UPLOAD_PATH}")" || true
+          cp "${ARTIFACT_ARCHIVE_NAME}" "./artifacts/upload/${ARTIFACT_UPLOAD_PATH}.zip"
+          mkdir -p "./artifacts/upload/${DOCS_UPLOAD_DIR}" || true
+          cp "${ARTIFACT_ARCHIVE_NAME}" "./artifacts/upload/${ARTIFACT_UPLOAD_PATH}.zip"
+        env:
+          TARGET_DIR: target/
+          ARTIFACT_UPLOAD_PATH: /imageflow-${{env.GITHUB_REF_NAME_SLUG}}-${{env.GITHUB_SHA_SHORT}}-${{matrix.name}}
+          ARTIFACT_ARCHIVE_NAME: ./artifacts/staging/archive.zip
+          DOCS_UPLOAD_DIR}: doc
+
+      - name: Nuget to S3
+        shell: bash
+        run: ./ci/pack_nuget/pack.sh tool
+        env:
+          CI_TAG: ${{env.GITHUB_REF_NAME_SLUG}}
+          PACKAGE_SUFFIX: ${{matrix.name}}
+          NUGET_RUNTIME: ${{matrix.NUGET_RUNTIME}}
+
+      - name: Upload to release (POSIX)
+        uses: Shopify/upload-to-release@master
+        if: ${{!contains( matrix.os, 'windows' )}}
+        with:
+          name: imageflow-${{env.GITHUB_REF_NAME_SLUG}}-${{env.GITHUB_SHA_SHORT}}-${{matrix.name}}.tar.gz
+          path: ./artifacts/upload/imageflow-${{env.GITHUB_REF_NAME_SLUG}}-${{env.GITHUB_SHA_SHORT}}-${{matrix.name}}.tar.gz
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          content-type: text/plain
+
+      - name: Upload to release (Windows)
+        uses: Shopify/upload-to-release@master
+        if: ${{contains( matrix.os, 'windows' )}}
+        with:
+          name: imageflow-${{env.GITHUB_REF_NAME_SLUG}}-${{env.GITHUB_SHA_SHORT}}-${{matrix.name}}.zip
+          path: ./artifacts/upload/imageflow-${{env.GITHUB_REF_NAME_SLUG}}-${{env.GITHUB_SHA_SHORT}}-${{matrix.name}}.zip
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          content-type: text/plain
+
+      - name: Upload to S3
+        uses: shallwefootball/s3-upload-action@master
+        with:
+          aws_key_id: ${{ secrets.AWS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY}}
+          aws_bucket: ${{ secrets.AWS_BUCKET }}
+          source_dir: ./artifacts/upload/
+          destination_dir: ''
+      - name: Upload to S3
+        uses: shallwefootball/s3-upload-action@master
+        with:
+          aws_key_id: ${{ secrets.AWS_KEY_ID }}
+          aws_secret_access_key: ${{ secrets.AWS_SECRET_ACCESS_KEY}}
+          aws_bucket: ${{ secrets.AWS_BUCKET }}
+          source_dir: ./artifacts/nuget/
```

Comment:
I see we uploaded the nuget package to S3, but we mostly need to upload it to nuget.org. You can see how we do this with CURL here: https://github.com/imazen/imageflow/blob/9277047ca3e3328d0fb421c4f48c8dfbfb5cf421/ci/travis_run.sh#L428-L443
You can test by pushing to Github's Packages system. The following packages must upload:

https://www.nuget.org/packages/Imageflow.NativeRuntime.win-x86/1.5.3-rc57
https://www.nuget.org/packages/Imageflow.NativeRuntime.win-x86_64/1.5.3-rc57
https://www.nuget.org/packages/Imageflow.NativeRuntime.osx_10_11-x86_64/1.5.2-rc56
https://www.nuget.org/packages/Imageflow.NativeRuntime.win-x86_64-haswell/1.5.3-rc57
https://www.nuget.org/packages/Imageflow.NativeRuntime.ubuntu_18_04-x86_64-haswell/1.5.3-rc57
https://www.nuget.org/packages/Imageflow.NativeRuntime.ubuntu_16_04-x86_64/1.5.3-rc57
https://www.nuget.org/packages/Imageflow.NativeRuntime.ubuntu_18_04-x86_64/1.5.3-rc57

Reply:
There are also .NativeTool equivalents for each of these that should automatically be created by pack_nuget.sh